Code/Framework/AzFramework/Platform/Linux/AzFramework/Application/Application_Linux.cpp

```diff
@@ -16,17 +16,30 @@ namespace AzFramework
     class ApplicationLinux
         : public Application::Implementation
         , public LinuxLifecycleEvents::Bus::Handler
+#if AZ_LINUX_WINDOW_MANAGER_XCB
+        , public LinuxXCBConnectionManager::Bus::Handler
+#endif // AZ_LINUX_WINDOW_MANAGER_XCB
```

Comment:
We can reduce the amount of `#ifdef` if we add the ApplicationLinux class contain class that implements the LinuxXCBConnectionManager interface.

i.e
```
if AZ_LINUX_WINDOW_MANAGER_XCB
class LinuxXCBConnectionManagerImpl
    : public LinuxXCBConnectionManager
{
        xcb_connection_t* GetXCBConnection() const override;
};
#endif // AZ_LINUX_WINDOW_MANAGER_XCB

class ApplicationLinux
    : ....
{
    LinuxXCBConnectionManagerInterface m_xcbConnectionManager;
}
```

Reply:
Then in the constructor and the destructor the ApplicationLinux Class, a `LinuxXCBConnectionManagerImpl` can be created and stored into the interface

```
    ApplicationLinux::ApplicationLinux()
    {
        LinuxLifecycleEvents::Bus::Handler::BusConnect();

#if AZ_LINUX_WINDOW_MANAGER_XCB
        m_xcbConnectionManager = AZStd::make_unique<LinuxXCBConnectionManagerImpl>();
        if (m_xcbConnectionManager..Get() == nullptr)
        {
            m_xcbConnectionManager.Register(m_xcbConnectionManager.get());
        }
#endif // AZ_LINUX_WINDOW_MANAGER_XCB
    }

    ApplicationLinux::~ApplicationLinux()
    {
#if AZ_LINUX_WINDOW_MANAGER_XCB
        if (m_xcbConnectionManager..Get() == m_xcbConnectionManager.get())
        {
            m_xcbConnectionManager.Unregister(m_xcbConnectionManager.get());
        }
        m_xcbConnectionManager.reset();
#endif // AZ_LINUX_WINDOW_MANAGER_XCB
        LinuxLifecycleEvents::Bus::Handler::BusDisconnect();
```

Reply:
Updated (with fixes)